enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,499 @@
+* Allow connection of multiple networks via explicit router API configuration.
+
+
+## Proposal
+
+By default in OVN-Kubernetes, pods are attached to what is known as the “default network”, which is a routed network divided up into a subnet per node. All pods will continue to have an attachment to this network, even when assigned a different primary network. Therefore, when a pod is assigned to a unique network, it will have two interfaces, one to the default network, and one to the unique network. The default network is required in order to provide:
+
+
+
+1. KAPI service access
+2. DNS service access
+3. Kubelet healthcheck probes to the pod
+
+All other traffic from the pod will be dropped by firewall rules on this network, when the pod is assigned a unique primary network. Routes will be added to the pod to route KAPI/DNS traffic out towards the default network. Note, it may be desired to allow access to any Kubernetes service on the default network (instead of just KAPI/DNS), but at a minimum KAPI/DNS will be accessible. Furthermore, the IP of the pod from the Kubernetes API will continue to show the IP assigned in the default network.
+
+In OVN-Kubernetes secondary networks are defined using Network Attachment Definitions (NADs). For more information on how these are configured, refer to:
+
+[https://github.com/ovn-org/ovn-kubernetes/blob/master/docs/features/multi-homing.md](https://github.com/ovn-org/ovn-kubernetes/blob/master/docs/features/multi-homing.md)
+
+The proposal here is to leverage this existing mechanism to create the network. The only new field that will be introduced to the NAD spec is a “primary” key which indicates that this network should be used for the pods primary network. For example, an OVN-Kubernetes NAD may look like:
+
+
+```
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: l3-network
+  namespace: ns1
+spec:
+  config: |2
+    {
+            "cniVersion": "0.3.1",
+            "name": "l3-network",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"layer3",
+            "subnets": "10.128.0.0/16/24",
+            "mtu": 1300,
+            "netAttachDefName": "ns1/l3-network"
+            "primaryNetwork": true
+    }
+```
+
+
+The NAD must be created before any pods are created for this namespace. Otherwise, pods will connect to the default primary network as their primary network. Only one non-default primary network may exist per namespace. A pod may not connect to multiple primary networks other than the default. Overlapping CIDRs may be used across NADs. When the NAD is created, OVN-Kubernetes will validate the configuration, as well as that no pods have been created, and then set the active network on the namespace via an annotation: [k8s.ovn.org/active-network](http://k8s.ovn.org/active-network): l3-network. If pods existed before the NAD was created, errors will be logged, and the active-network will remain “default”.
```

Comment:
I'd rather prevent the pod from starting, instead of silently starting it in a network different from the one associated to the current config.

Reply:
That seems reasonable. We can block pod creation in the namespace if it has more than one primary network defined. I'll add that.

Reply:
I updated the text.